### PR TITLE
[BIOMAGE-1494] - Simplify filter disabled error messages

### PR DESCRIPTION
**Pull request by aerlaut**
# Background
#### Link to issue 

https://biomage.atlassian.net/browse/BIOMAGE-1494

#### Link to staging deployment URL 

https://ui-agi-prefiltered.scp-staging.biomage.net/

#### Links to any Pull Requests related to this

UI - https://github.com/biomage-ltd/ui/pull/528
API - https://github.com/biomage-ltd/api/pull/242
Pipeline - https://github.com/biomage-ltd/pipeline/pull/178
IAC Migration - https://github.com/biomage-ltd/iac/pull/191

#### Anything else the reviewers should know about the changes here

- Simplify code that was checking for prefiltered samples in the UI. Now we use the prop directly to determine if the sample is prefiltered or not.
- Remove code in Data Integration which is not used anymore.
- A migration has been carried out to add `prefiltered` to existing experiments and enable this feature. 

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [X] Unit tests written
- [X] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [X] Deployed to staging
- [ ] Passed integration tests 

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [x] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: src/__test__/components/data-processing/data-integration/DataIntegration.test.jsx [hunk 1]
```diff
@@ -32,14 +32,6 @@ const mockStore = configureStore([thunk]);
...
 
-jest.mock('next/router', () => ({
-  useRouter: jest.fn().mockImplementation(() => ({
-    query: {
-      experimentId: '1234',
-    },
-  })),
-}));
-
 const mockedStore = mockStore({
```
Comment (ivababukova) [inline src/__test__/components/data-processing/data-integration/DataIntegration.test.jsx:35]: why was this deleted?

File: src/__test__/pages/experiments/[experimentId]/data-processing/index.test.jsx [hunk 2]
```diff
@@ -1,41 +1,71 @@
...
+
+jest.mock('../../../../../components/UserButton', () => () => <></>);
+
+// Mock all filter components
+jest.mock('../../../../../components/data-processing/CellSizeDistribution/CellSizeDistribution', () => () => <></>);
+jest.mock('../../../../../components/data-processing/MitochondrialContent/MitochondrialContent', () => () => <></>);
+jest.mock('../../../../../components/data-processing/GenesVsUMIs/GenesVsUMIs', () => () => <></>);
+jest.mock('../../../../../components/data-processing/DoubletScores/DoubletScores', () => () => <></>);
+jest.mock('../../../../../components/data-processing/DataIntegration/DataIntegration', () => () => <></>);
+jest.mock('../../../../../components/data-processing/ConfigureEmbedding/ConfigureEmbedding', () => () => <></>);
```
Comment (ivababukova) [inline src/__test__/pages/experiments/[experimentId]/data-processing/index.test.jsx]: you no longer need to add `.../.../` to these, the linter should understand the relative imports

File: src/__test__/components/data-processing/data-integration/DataIntegration.test.jsx (hunk 1 above)
Comment (aerlaut) [reply src/__test__/components/data-processing/data-integration/DataIntegration.test.jsx:35]: This is not used anymore in Data Integration. It was previously used to trigger an alert which shows that there are changes to Data Integration that has not been changed, and whether the user would like to run or discard the changes. 

Now we can make changes to multiple parts of the pipeline before running the pipeline again. So, this code is not needed anymore.

File: src/__test__/pages/experiments/[experimentId]/data-processing/index.test.jsx (hunk 2 above)
Comment (aerlaut) [reply src/__test__/pages/experiments/[experimentId]/data-processing/index.test.jsx]: removed

File: src/__test__/pages/experiments/[experimentId]/data-processing/index.test.jsx [hunk 3]
```diff
@@ -1,41 +1,71 @@
 import React from 'react';
-import { mount, configure } from 'enzyme';
+import { screen, render } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import userEvent from '@testing-library/user-event';
+// import { mount, configure } from 'enzyme';
```
Comment (ivababukova) [inline src/__test__/pages/experiments/[experimentId]/data-processing/index.test.jsx]: remove this commented out line

File: src/__test__/test-utils/experimentSettings.mock.js [hunk 4]
```diff
@@ -14,6 +14,16 @@ const sampleifiedConfig = (sampleIds, configToReplicate) => {
 };
 
 const generateProcessingConfigMock = (sampleIds) => ({
+  classifier: {
```
Comment (ivababukova) [inline src/__test__/test-utils/experimentSettings.mock.js:17]: what about the prefiltered parameter?

Comment (aerlaut) [reply src/__test__/test-utils/experimentSettings.mock.js:17]: oh right. If `enabled` is true, then `prefiltered` is false.

**Review by ivababukova — APPROVED** [review]

File: src/__test__/test-utils/experimentSettings.mock.js (hunk 4 above)
Comment (aerlaut) [reply src/__test__/test-utils/experimentSettings.mock.js:17]: Added
